include/sta/ArrayTable.hh

```diff
@@ -58,19 +58,16 @@ public:
 
 private:
   ArrayBlock<TYPE> *makeBlock(uint32_t size);
-  void pushBlock(ArrayBlock<TYPE> *block);
+  void pushBlock(size_t size);
   void deleteBlocks();
 
   size_t size_;
   // Block index of free block (blocks_[size - 1]).
   BlockIdx free_block_idx_;
   // Index of next free object in free_block_idx_.
   ObjectIdx free_idx_;
-  // Don't use std::vector so growing blocks_ can be thread safe.
```

Comment:
I guess you missed the comment that you deleted that explains why std::vector cannot be used here. If you look at the commit history you will see that it used to use std::vector. When the the vector grows it fails with multiple threads.

Reply:
I didn't miss it. `vector::swap` is basically no different from the previous approach. See https://godbolt.org/z/8sqqTPMMf

However, I suppose implementations other than libstdc++ might exhibit different behavior on lower optimization levels. So I pushed a manual swap implementation just to be sure.